src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/IndexController.php

```diff
@@ -38,4 +50,116 @@ public function tableAction(): JsonResponse
 
         return $this->jsonResponse($table->fetchData());
     }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     *
+     * @return array|\Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function createAction(Request $request)
+    {
+        $punchoutCatalogConnectionsForm = $this->getFactory()
+            ->getPunchoutCatalogConnectionForm();
+
+        $punchoutCatalogConnectionsForm->handleRequest($request);
+
+        if ($punchoutCatalogConnectionsForm->isSubmitted() && $punchoutCatalogConnectionsForm->isValid()) {
+            $punchoutCatalogResponseTransfer = $this->getFacade()
+                ->createConnection($punchoutCatalogConnectionsForm->getData());
+
+            if ($punchoutCatalogResponseTransfer->getIsSuccessful()) {
+                $this->addSuccessMessage('Connection added');
+            }
+
+            $this->handleResponseErrors($punchoutCatalogResponseTransfer);
+
+            return $this->redirectResponse(static::ROUTE_PUNCHOUT_CATALOGS_CONNECTION_CREATE_PAGE);
+        }
+
+        return [
+            'punchoutCatalogConnectionForm' => $punchoutCatalogConnectionsForm->createView(),
+        ];
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     *
+     * @return array|\Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function editAction(Request $request)
+    {
+        $idPunchoutCatalogConnection = $this->castId(
+            $request->query->get(static::ROUTE_PARAM_ID_PUNCHOUT_CATALOG_CONNECTION)
+        );
+
+        $punchoutCatalogConnectionTransfer = $this->getFacade()
+            ->findConnectionById($idPunchoutCatalogConnection);
+
+        if (!$punchoutCatalogConnectionTransfer) {
+            $this->addErrorMessage('Connection not found');
+
+            return $this->redirectResponse(static::ROUTE_PUNCHOUT_CATALOGS_CONNECTION_LIST_PAGE);
+        }
+
+        $punchoutCatalogConnectionsForm = $this->getFactory()
+            ->getPunchoutCatalogConnectionForm($punchoutCatalogConnectionTransfer);
+
+        $punchoutCatalogConnectionsForm->handleRequest($request);
+
+        if ($punchoutCatalogConnectionsForm->isSubmitted() && $punchoutCatalogConnectionsForm->isValid()) {
+             return $this->processConnectionEditForm($punchoutCatalogConnectionsForm);
+        }
+
+        return [
+            'punchoutCatalogConnectionForm' => $punchoutCatalogConnectionsForm->createView(),
+        ];
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormInterface $punchoutCatalogConnectionsForm
+     *
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    protected function processConnectionEditForm(FormInterface $punchoutCatalogConnectionsForm): RedirectResponse
```

Comment:
This method name is `processConnectionEditForm`.
Its argument name is `$punchoutCatalogConnectionsForm`.

For me, they feel like they aren't connected with each other.
Are you sure, this is the best naming for this case?